libvncclient/rfbproto.c

```diff
@@ -1585,6 +1591,52 @@ SendKeyEvent(rfbClient* client, uint32_t key, rfbBool down)
   return WriteToRFBServer(client, (char *)&ke, sz_rfbKeyEventMsg);
 }
 
+/*
+ * Resize client
+ */
+rfbBool
+ResizeClientBuffer(rfbClient* client, int width, int height)
+{
+  client->width = width;
+  client->height = height;
+  client->updateRect.x = client->updateRect.y = 0;
+  client->updateRect.w = client->width;
+  client->updateRect.h = client->height;
+  return client->MallocFrameBuffer(client);
+}
+
+/*
+ * SendExtDesktopSize
+ */
+
+rfbBool
+SendExtDesktopSize(rfbClient* client, uint16_t width, uint16_t height)
+{
+  rfbSetDesktopSizeMsg sdm;
+  if (client->screen.width == 0 && client->screen.height == 0 ) {
+    rfbClientLog("Screen not yet received from server - not sending dimensions %dx%d\n", width, height);
+    return TRUE;
+  }
+  if (client->screen.width != rfbClientSwap16IfLE(width) && client->screen.height != rfbClientSwap16IfLE(height)) {
```

Comment:
Not sure why rfbClientSwap16IfLE is needed here, as width and height are not going over the network. Can you please explain?

Reply:
Hello @bk138
I did some tests and I think using rfbClientSwap16IfLE here is correct

Debug variables values:
client->screen.width: 21764
client->screen.height: 16131
width: 1109
height: 831
rfbClientSwap16IfLE(width): 21764
rfbClientSwap16IfLE(height): 16131